src/main/webapp/app/overview/notification/notification.service.ts

```diff
@@ -183,4 +174,13 @@ export class NotificationService {
         const courseId = target.course || notification.course.id;
         this.router.navigate([target.mainPage, courseId, target.entity, target.id]);
     }
+
+    private initNotificationObserver(): void {
+        this.notificationObserver = new BehaviorSubject<Notification | null>(null);
+    }
+
+    public cleanUp(): void {
+        this.cachedNotifications = new Observable<EntityArrayResponseType>();
```

Comment:
I guess we should cleanup `subscribedTopics` as well, right?

Reply:
Had that on my todo :-) Is now cleaned up during logout: https://github.com/ls1intum/Artemis/pull/1323/commits/0af1890947ced7227abc9f4e7db24fe3ed1a5505